.travis.yml

```diff
@@ -1,40 +1,26 @@
-language: c
+language: generic
 os: linux
-sudo: required
-matrix:
-  include:
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-            - george-edison55-precise-backports
-          packages:
-            - gcc-7
-            - gfortran-7
-            - g++-7
-            - cmake-data
-            - cmake
-      env:
-        - MATRIX_EVAL="FC=gfortran-7 && CC=gcc-7 && CXX=g++-7"
-        - GCC=7
+sudo: false
+dist: trusty
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+      - george-edison55-precise-backports
+    packages:
+      - gfortran-7
+      - cmake-data
+      - cmake
+    env:
+      - GCC=7
 
 before_install:
   - export FC=gfortran-${GCC}
-  - export CC=gcc-${GCC}
-install:
-#  - sudo apt-get -qq update
-  - sudo apt-get install -y libxml2-dev
-  - sudo apt-get install -y lcov curl
-  - sudo apt-get install gfortran
-  - sudo apt-get install cmake
-compiler:
-  - gcc
 script:
-  - mkdir build && cd build && cmake ../ && make -j2 && ctest -j2 
+  - mkdir build && cd build && cmake ../ && make -j $(nproc) && ctest -j2
 after_success:
   - gcov-${GCC} --version
-  - bash <(curl -s https://codecov.io/bash) -t 8fd6bb24-0eac-4adf-a45a-d68bf843203c
```

Comment:
In general, you shouldn't have tokens in clear text. You can use the `travis` ruby gem to help you create [encrypted variables](https://docs.travis-ci.com/user/environment-variables#sts=Defining encrypted variables in .travis.yml), but even with these, sometimes they can "leak" in the travis-ci logs if you're not careful and they get printed to the screen due to an error. The codecov token isn't super sensitive, because worse case scenario someone can post phony coverage data. But in general you should avoid using tokens that aren't encrypted and take a lot of precautions when you do use encrypted tokens on Travis-ci. I have had GH tokens "leak" in the logs which means that someone could potentially gain access to private code depending on the permissions on the token.

For this case, Codecov.io integrates natively with Travis-CI.org so you don't need to have a Codecov.io token at all. For Travis-CI.com (private) you may need to provide a token, and if you're sending coverage data from local testing or some other private CI server you will need the token.

I would recommend [resetting the token on Codecov.io](https://codecov.io/gh/porteri/vtkmofo/settings) since this one has been made public.